primitives/api/src/lib.rs

```diff
@@ -350,21 +350,39 @@ pub trait ApiErrorExt {
 	type Error: std::fmt::Debug + From<String>;
 }
 
+/// Describes on what should happen with a transaction.
+pub enum TransactionOutcome<R> {
+	/// Commit the transaction.
+	Commit(R),
+	/// Rollback the transaction.
+	Rollback(R),
+}
```

Comment:
We already have that in `frame_support`. Maybe you can `pub use` this in `frame_support` or move this to some common location where both can import this.